types/lodash/common/lang.d.ts

```diff
@@ -572,10 +572,10 @@ declare module "../index" {
          * @return Returns true if value is empty, else false.
          */
         isEmpty<T extends { __trapAny: any }>(value?: T): boolean;
-        isEmpty(value: string): value is '';
-        isEmpty(value: Map<any, any> | Set<any> | List<any> | null | undefined): boolean;
+        isEmpty(value: any): value is '' | null | undefined | 0 | false;
```

Comment:
to my surprise, at least in TS sandbox, it's possible to extend this with `| []` and it will work as expected:

```
const a: number[] = []
//    ^? const a: number[]
if (Math.random()) {
    a.push(11)
}
if (isEmpty(a)) {
    const e = a
//        ^? const e: []
}
```

are there hidden risk with using `is []`?

Reply:
It's tricky because I think array types should not be filtered out whether this function returns true or false. So it seems like both options are wrong... Or do you think you can find a way to make that work?

Reply:
>  I think array types should not be filtered out whether this function returns true or false. So it seems like both options are wrong... 

I'm sorry, absolutely did not get it :) could you put some small illustration?

Reply:
I probably got what do you mean here. have just found that `| object` does not disappear in `else` branch, so the same as to object, with arrays my code will not work well 🙄 

Looking for a solution for both.

Reply:
it seems, I understood "function overload vs union" wrong